### PR TITLE
Added basic integration tests and fixed related bugs

### PR DESCRIPTION
**Pull request by rpyzh**
Also modified `update_note` endpoint to be able to set `is_public` attribute.

### CODE REVIEW

File: app_test.py [hunk 1]
```diff
@@ -73,6 +74,77 @@ def test_update_note_with_invalid_token(self):
...
+
+        self.assertEqual(add_resp.status_code, 200)
+        self.assertEqual(note_response_payload["note"], note_payload["note"])
+        self.assertEqual(note_response_payload["owner_id"], _id)
+
+        # Updating phase
+        update_payload = {"note_id": _note_id, "note": "updated test note"}
+        upd_resp = self.app.post("/update_note", headers=headers, json=update_payload)
+        update_response_payload = json.loads(upd_resp.get_json())
+        print(update_response_payload)
```
Comment (ss14635) [inline app_test.py:104]: nit: remove or comment out this print statement

File: app_test.py [hunk 2]
```diff
@@ -73,6 +74,77 @@ def test_update_note_with_invalid_token(self):
...
+        self.assertEqual(upd_resp.status_code, 200)
+        self.assertEqual(update_response_payload["status"], "success")
+
+    def test_integration_register_post_set_public(self):
+        _name = "Roman P"
+        _id = "roman@roman.com"
+
+        # Registration phase
+        reg_resp = self.app.post("/register", json={"id": _id, "name": _name})
+        print(reg_resp.get_json())
```
Comment (ss14635) [inline app_test.py:115]: nit: remove or comment out this print statement

File: app_test.py [hunk 3]
```diff
@@ -73,6 +74,77 @@ def test_update_note_with_invalid_token(self):
...
+
+        self.assertEqual(add_resp.status_code, 200)
+        self.assertEqual(note_response_payload["note"], note_payload["note"])
+        self.assertEqual(note_response_payload["owner_id"], _id)
+
+        # Updating phase
+        update_payload = {"note_id": _note_id, "is_public": True}
+        upd_resp = self.app.post("/update_note", headers=headers, json=update_payload)
+        update_response_payload = json.loads(upd_resp.get_json())
+        print(update_response_payload)
```
Comment (ss14635) [inline app_test.py:136]: nit: remove or comment out this print statement

File: main.py [hunk 4]
```diff
@@ -56,28 +56,30 @@ def home():
...
+
+    return jsonify(user1.toJson())
 
 
 @app.route("/add_note", methods=["POST"])
 @token_required
 def add_note(current_user):
     print("current user - " + json.dumps(current_user.__dict__))
     content = request.json
+    print(content)
```
Comment (ss14635) [inline main.py:75]: nit: remove or comment out this print statement

**Review by ss14635 — APPROVED** [review]
I haven't tested, but I assume the integration tests are passing here, so I'll approve.

File: main.py (hunk 4 above)
Comment (rpyzh) [reply main.py:74]: not touching main code debug at this point

File: app_test.py (hunk 3 above)
Comment (rpyzh) [reply app_test.py]: Taken care of. Thanks!

File: app_test.py (hunk 2 above)
Comment (rpyzh) [reply app_test.py]: Taken care of. Thanks!

File: app_test.py (hunk 1 above)
Comment (rpyzh) [reply app_test.py]: Taken care of. Thanks!

**Review by ajit-shrivastav-nyu — APPROVED** [review]

**Pull request merged**